### ISSUE TITLE
🚀 Release v1.1.0

### ISSUE THREAD

**Issue by davidgatti** [0x4447-aws-products-cfn/0x4447_product_paid_sftp_single_user#3]
## Steps to release the new product

- [x] Log in to AWS
- [x] Assume the Marketplace account
- [x] Open new tab and visit the [Marketplace Portal](https://aws.amazon.com/marketplace/management/homepage).
- [x] From the top menu bar, click Assets, and from the drop down, select Amazon Machine Image.
- [x] Click the orange button, that say Add AMI.
- [x] From the table select the image you want to release, and click next.
- [x] On the next page in the `IAM Access role ARN` paste this: `arn:aws:iam::239748505547:role/aws_marketplace` and click Next.
- [x] On this page there is nothing to do, so you can click Next.
- [x] On the summary page check that all is OK, and click Submit.

Once this process is done, on the Assets > Amazon Machine Images (AMIs) page you will be able to see the status of the scan. you have to wait for the scan to be completed, some time it takes 1h or more.

Once the scan is completed, you can now release the product. By doing the following:

- [x] In the top menu, click Products, and from the drop down, select Server
- [x] From the list select the product that you want to update.
- [x] On the new page, next to the name of the products, you will see a bunch of buttons and drop downs.
- [x] Select Request Change
  - [x] From the drop down, select: Update versions
  - [x] Then select Add new version.
- [x] On the new page you have to change the following:
  - [x] Version title: write the version.
  - [x] Release notes ( copy them from: [](https://www.notion.so/ba8f459abcef479187409dc9fc0713ff))
  - [x] Amazon machine image (AMI) ID ( the once the you scanned)
  - [x] IAM access role ARN ( the same as you provided above)
  - [x] And that is it on this page
- [x] On the next page don’t change anything, click on `Skip to Review and submit`
- [x] On the next page click Submit

Once the review is OK, and released by AWS, we have to update the CloudFormation file with the latest AMIs of the product for each region, to do so you have to run a Lambda function that will pull all of this information automatically for you.

- [x] Go to the Lambda dashboard and run [ami_id_extractor](https://us-east-1.console.aws.amazon.com/lambda/home?region=us-east-1#/functions/ami_id_extractor?tab=testing).
- [x] Scroll down until you see the `Event JSON` section.
- [x] Replace the `id` and the `ami_name` data:
  - [x] How to get the `id`: click this [link](https://aws.amazon.com/marketplace/management/products/server?#) > select the product that you are updating > in the `Product summary` section you will see: `Product ID`
  - [x] How to get the `ami_name`: click this [link](https://us-east-1.console.aws.amazon.com/ec2/home?region=us-east-1#Images:visibility=owned-by-me) > from the table with all the ami, look for the `AMI name` column, and copy and past the name of the product that you are working on.
- [x] Run the Lambda
- [x] In the output you will have a JSON that you have to copy and paste in the product CloudFormation file.
- [x] In the GitHub account: [https://github.com/0x4447-aws-products-cfn](https://github.com/0x4447-aws-products-cfn) you will have all the main CFNs, pick the right product and clone it:
- [x] On your local machine open your favorite editor and navigate to
  - [x] 04_Mappings - folder
  - [x] os_id.json - file
  - [x] Replace the content from here with what the Lambda from above outputted.
- [x] Install the [grapes](https://www.npmjs.com/package/@0x4447/grapes) module from NPM.
- [x] Then in the folder where the repo is, run `grapes -s .` this will regenerate the CloudFormation file with the new values.
- [x] Now in the folder you will have a `CloudFormation.json` is created, use this file to deploy the stack in AWS for testing. Test the the new image is working as expected, and if all is OK delete the stack, and…
- [x] Make a new Pull Request (the PR should only have the changes of the ID and nothing else)
- [x] Mention me in the PR so I get a notification.
- [x] Merge it with production (this will kick a auto deployment)

And this is it when it comes to releasing, the last thing to do is to check if the official CFN works with the new product. For this do the following:

- [x] Go to the products page, and pick the product that you are releasing: [https://products.0x4447.com/](https://products.0x4447.com/)
- [x] Click on the documentation page of the selected product
- [x] Scroll until the Deployment section
- [x] Open the source JSON file, and verify that the new ID are in place.
- [x] In this section there is a link to the RAW CFN file, click it and verify that the new AMI IDs got updated.
- [x] Go back.
- [x] And try to deploy the product using the official CFN
- [x] Once the product is deployed, check that the correct version got deployed
- [x] Test the main feature of the product
- [x] Delete the stack when finished.
- [x] Go to the EC2 Dashboard
- [x] In the AMI section, rename the AMI to make it the official image with the right version if necessary.

And this is the final step, if all is OK, then the process of releasing is done.

**Comment by davidgatti** [comment]
# Release note.

- Added support for individual file encryption at rest.

**Comment by vladpona** [comment]
All steps are done. All works as expected!
